packages/flutter_tools/lib/src/base/async_guard.dart

```diff
@@ -99,9 +99,9 @@ Future<T> asyncGuard<T>(
       completer.completeError(e, s);
       return;
     }
-    if (onError is _BinaryOnError) {
+    if (onError is _BinaryOnError<T>) {
```

Comment:
I think this should probably be `_BinaryOnError<Null>` and `_UnaryOnError<Null>`, the T there is unrelated to the overall value. @zanderso

Reply:
`onError` is used to complete a `Completer<T>` so I think `_BinaryOnError<T>` is the right type to check here.

Reply:
Ahh yes you're right!